File: SundeeFundee/Sources/SundeeFundeeKit/DomainLayer/ExerciseValue.swift
import Foundation

// MARK: - Exercise Value (Discriminated Union)

/// A discriminated union for sets/reps values in program templates.
/// Matches the TypeScript ExerciseValue type.
public enum ExerciseValue: Equatable, Codable, Sendable {
    case fixed(value: Int)
    case amrap
    case range(low: Int, high: Int)
    case text(value: String)

    public var description: String {
        switch self {
        case .fixed(let value): return "\(value)"
        case .amrap: return "AMRAP"
        case .range(let low, let high): return "\(low)\u{2013}\(high)"
        case .text(let value): return value
        }
    }
}

File: android/core/src/main/java/com/sundeefundee/core/domain/exercise/ExerciseValue.kt
package com.sundeefundee.core.domain.exercise

import kotlinx.serialization.Serializable

// Discriminated union for set types — mirrors iOS ExerciseType
@Serializable
sealed class ExerciseValue {
    @Serializable
    data class Fixed(val value: Int) : ExerciseValue()

    @Serializable
    data object Amrap : ExerciseValue()

    @Serializable
    data class Range(val min: Int, val max: Int) : ExerciseValue()

    @Serializable
    data class Text(val value: String) : ExerciseValue()

    val description: String
        get() = when (this) {
            is Fixed -> "Fixed"
            is Amrap -> "AMRAP"
            is Range -> "$min-$max reps"
            is Text -> value
        }
}
